airbyte-config/init/src/main/java/io/airbyte/config/init/RemoteConnectorCatalogPersistence.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.init;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.util.MoreIterators;
+import io.airbyte.config.AirbyteConfig;
+import io.airbyte.config.AirbyteConfigValidator;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.ConfigWithMetadata;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigPersistence;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This config persistence pulls the connector configurations from a remotely hosted catalog. It is
+ * read-only.
+ */
+final public class RemoteConnectorCatalogPersistence implements ConfigPersistence {
```

Comment:
It this the right interface for us to be using? I think we used this back in a time on when we had a lot less clarity on what our goals are--it was partially a trick to get platform upgrades to work. Now we have a much more focused version on "catalog persistence". Using this very broad interface means we lose a lot of clarity. As we are refactoring this, it would be a good moment to revisit what this iface should be. 

cc @evantahler

Reply:
@cgardens, I don't get if you suggest I use a different already existing interface or if we need to create a new, more focused one.

Reply:
The second option.

Reply:
+1 on the second option.

In terms of interface, I think what's being abstracted here is the `source_definitions` and `destination_definitions`.
Probably an `ActorDefinitionsProvider` with a `listDefinitions` method could work.
The current behavior could be wrapped in a `SeedResource` implementation while your new one would be a `RemoteResource`.

Injecting this into the the existing implementations would help us avoid adding a new layer of ConfigPersistence and a step forward into breaking this giant into more focused pieces.

Reply:
I had initially suggested implementing a new `ConfigPersistence` class since that's what we were using for the `YamlSeedConfigPersistence`, which is doing a similar thing, and would be easy to slot in as a replacement when [loading in the definitions for cloud](https://github.com/airbytehq/airbyte-cloud/blob/c5fed837c3a76beb5f7392ec1b67e93cc4e7e6e6/airbyte-bootloader-wrapped/src/main/java/io/airbyte/bootloader/wrapped/BootloaderAppWrapped.java#L91). But I definitely agree that the typing guarantees and how generic ConfigPersistence is isn't ideal... If we want to move away from it anyway, this being an `ActorDefinitionsProvider` like jimmy proposed sounds great to me.